src/python/src/pyclient.cpp

```diff
@@ -1113,4 +1114,237 @@ void PyClient::save(std::vector<std::string> addresses)
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing get_list_length.");
+    }
+}
+
+// Poll list length until length is equal
+bool PyClient::poll_list_length(const std::string& name, int list_length,
+                                int poll_frequency_ms, int num_tries)
+{
+    try {
+        return _client->poll_list_length(
+            name, list_length, poll_frequency_ms, num_tries);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing poll_list_length.");
+    }
+}
+
+// Poll list length until length is greater than or equal
+bool PyClient::poll_list_length_gte(const std::string& name, int list_length,
+                                    int poll_frequency_ms, int num_tries)
+{
+    try {
+        return _client->poll_list_length_gte(
+            name, list_length, poll_frequency_ms, num_tries);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing poll_list_length_gte.");
+    }
+}
+
+// Poll list length until length is less than or equal
+bool PyClient::poll_list_length_lte(const std::string& name, int list_length,
+                                   int poll_frequency_ms, int num_tries)
+{
+    try {
+        return _client->poll_list_length_lte(
+            name, list_length, poll_frequency_ms, num_tries);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing poll_list_length_lte.");
+    }
+}
+
+// Get datasets from an aggregation list
+py::list PyClient::get_datasets_from_list(const std::string& list_name)
+{
+    try {
+        std::vector<DataSet> datasets = _client->get_datasets_from_list(list_name);
+        std::vector<PyDataset*> result;
+        for (auto it = datasets.begin(); it != datasets.end(); it++) {
+            DataSet* ds = new DataSet(*it);
```

Comment:
Can we use move semantics here to avoid an extra underlying copy of all of the data.  It looks like we are just copying to extend the memory lifetime?n. This comment would extend to the other ``get`` method for aggregation lists.